integration/vm_factory/vm_templating_test.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 HyperHQ Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This test will perform several tests to validate kata containers with
+# vm templating.
+
+set -e -x
+
+cidir=$(dirname "$0")
+
+source "${cidir}/../../metrics/lib/common.bash"
+
+# Environment variables
+IMAGE="${IMAGE:-busybox}"
+CONTAINER_NAME=$(random_name)
+PAYLOAD_ARGS="${PAYLOAD_ARGS:-tail -f /dev/null}"
+
+# Timeout is the duration of this test (seconds)
+# We want to stress the agent for a significant
+# time (approximately running for two days)
+timeout=186400
+start_time=$(date +%s)
+end_time=$((start_time+timeout))
+
+if [ -f ${DEFAULTCONFIG} ]; then
+	runtime_config_path=${DEFAULTCONFIG}
+else
+	runtime_config_path=${PKGDEFAULTCONFIG}
+fi
+
+template_tmpfs_path="/run/vc/vm/template"
+
+enable_vm_template_config() {
+	echo "enable vm template config ${runtime_config_path}"
+	sudo sed -i -e 's/^#\(enable_template\).*=.*$/\1 = true/g' "${runtime_config_path}"
+}
+
+disable_vm_template_config() {
+	echo "disable vm template config"
+	sudo sed -i -e 's/^\(enable_template\).*=.*$/#\1 = true/g' "${runtime_config_path}"
+}
+
+init_vm_template() {
+	echo "init vm template"
+	sudo -E PATH=$PATH kata-runtime factory init
+}
+
+destroy_vm_template() {
+	echo "destroy vm template"
+	sudo -E PATH=$PATH kata-runtime factory destroy
+}
+
+check_vm_template_factory() {
+	echo "checking vm template factory"
+	mount | grep tmpfs | grep ${template_tmpfs_path}
+	res=$(ls ${template_tmpfs_path} | grep memory |wc -l)
+	test $res -eq 1
+	res=$(ls ${template_tmpfs_path} | grep state |wc -l)
+	test $res -eq 1
+}
+
+setup() {
+	clean_env
+}
+
+check_qemu_for_vm_template() {
+	echo "checking qemu command line for vm template arguments"
+	ps aux | grep qemu | grep ${template_tmpfs_path} | wc -l
+}
+
+test_create_container_with_vm_template() {
+	docker run --runtime=$RUNTIME -d --name $CONTAINER_NAME $IMAGE $PAYLOAD_ARGS
```

Comment:
use `sudo` for docker in case the tests is run using a user that is not part of the docker group? Same for all other docker calls below.